pkg/server/commands/check_command.go

```diff
@@ -28,6 +26,30 @@ const (
 	defaultMaxConcurrentReadsForCheck = math.MaxUint32
 )
 
+type InvalidTupleError struct {
+	Message string
+}
+
+func (e *InvalidTupleError) Error() string {
+	return e.Message
+}
+
+type InvalidRelationError struct {
+	Message string
+}
+
+func (e *InvalidRelationError) Error() string {
+	return e.Message
+}
+
+type ThrottledError struct {
+	Message string
+}
+
+func (e *ThrottledError) Error() string {
+	return e.Message
+}
+
```

Comment:
**Question:** I wish I had an answer for this but how does this compare to `/pkg/server/errors`? It seems like there is at least some overlap that can be consolidated. 

Errors have been a sore spot in the past so not asking for a complete overhaul but can you see if there can be some consolidation or reorganization?

Reply:
I was also unsatisfied with this setup, so I changed it significantly in the most recent commit. Probably easier to talk over zoom but here were my thoughts:

The server errors are specifically design to be [external facing](https://github.com/openfga/openfga/blob/main/pkg/server/errors/errors.go#L1) grpc errors. They build errors based on status codes and messages and then convert them to a pure error with a message like "rpc error: code = Code(2000) desc = oh no", and there's no way to backport that without matching on the string itself.

Many of our packages have their own separate error definitions, conditions, typesystem, storage. `Check` feels like a first-class citizen in our code, so it'd make sense to have its own specific error definitions, but our `package check` [is a test package](https://github.com/openfga/openfga/blob/main/tests/check/check.go#L1) for some reason.

[This commit](https://github.com/openfga/openfga/pull/2044/commits/df0101676da6ff10799b7a4c3f93101d3b46974b) creates an `errors.go` within the /commands package which absorbs all of the new error types and translation to GRPC that was scattered around in the PR from last week. I'm not 100% wed to this setup but it does feel better than stuffing more cruft into the server file.